File: abc132/abc132_a.cs
// https://atcoder.jp/contests/abc132/submissions/8272517

using System;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var input = Console.ReadLine();
            if (input[0] == input[1] && input[1] != input[2] && input[2] == input[3]
                || input[0] == input[2] && input[2] != input[1] && input[1] == input[3]
                || input[0] == input[3] && input[3] != input[1] && input[1] == input[2])
            {
                Console.WriteLine("Yes");
            }
            else
            {
                Console.WriteLine("No");
            }
        }
    }
}

File: abc/abc132/abc132_a.fs
// https://atcoder.jp/contests/abc132/submissions/11967575

stdin.ReadLine().ToCharArray()
|> Array.sort
|> fun S ->
    if S.[0] = S.[1] && S.[1] <> S.[2] && S.[2] = S.[3] then "Yes"
    else "No"
|> stdout.WriteLine
